include/ts/ts.h

```diff
@@ -962,6 +967,7 @@ tsapi TSMLoc TSMimeHdrFieldGet(TSMBuffer bufp, TSMLoc hdr, int idx);
 
  */
 tsapi TSMLoc TSMimeHdrFieldFind(TSMBuffer bufp, TSMLoc hdr, const char *name, int length);
+tsapi TSReturnCode TSMimeHdrFieldFastFind(TSMBuffer bufp, TSMLoc hdr, const char *name, int length, TSMLoc *locp);
```

Comment:
I suggest adding this to apidefs.h.in:
```
// WARNING:  Contents invalidated by calls to TSHttpTxnReenable().
typedef struct {
  void * _d[3];  // private
} TSHdrHandle;

#define TS_MLOC_FROM_HDR_HANDLE(HH_) ((TSMLoc *)((HH_)._d))
```
and changing the last parameter like this:
```
tsapi TSReturnCode TSMimeHdrFieldFastFind(TSMBuffer bufp, TSMLoc hdr, const char *name, int length, TSHdrHandle *);
```
And in InkAPI.cc adding:
```
static_assert(sizeof(TSHdrHandle) >= sizeof(MIMEFieldSDKHandle) && alignof(TSHdrHandle) == alignof(MIMEFieldSDKHandle))
```
for greater type safety / static checking for mistakes.

Reply:
Seems reasonable, although it should be `TSFieldHandle` to match the function name. A few points
1. Is `void *` the best choice? Why not `unsigned char` to indicate a raw buffer? Is this to help with alignment?
2. I think even in C one can return a struct by value. That might be a better choice than passing a pointer. The caller would then check the return value for validity rather than an explicit error code.

Reply:
@SolidWallOfCode I pushed a new commit with returning `TSHdrHandle`. Please take another look.